lib/mix/tasks/phoenix.gen.html.ex

```diff
@@ -66,7 +66,8 @@ defmodule Mix.Tasks.Phoenix.Gen.Html do
     end
   end
 
-  defp validate_args!([_, plural | _] = args) do
+  defp validate_args!([singular, plural | _] = args) do
+    Mix.Phoenix.check_module_name_availability!(singular)
```

Comment:
Sounds like this check should only occur in the Model generator where it is not appended with `View`, `Controller`, or `Channel`.  Thoughts?